.github/workflows/checks.yml

```diff
@@ -0,0 +1,96 @@
+name: Basic checks
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+concurrency:
+  # Only run once for latest commit per ref and cancel other (previous) runs.
+  group: ci-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check-code-style:
+    name: Check Code Style
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # we don't know what commit the last tag was it's safer to get entire repo so previousStableVersion resolves
+          fetch-depth: 0
+
+      - name: Set up JDK 11
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: adopt@1.11.0-11
+
+      - name: Cache Coursier cache
+        uses: coursier/cache-action@v6.3
+
+      - name: Code style check and binary-compatibility check
+        # Run locally with: sbt 'verifyCodeStyle ; mimaReportBinaryIssues'
+        # TODO: add versionSyncCheck when running CRON jobs
+        run: |
```

Comment:
I think we can split some tasks for parallels execution to decrease check time.
([play-json](https://github.com/playframework/play-json/blob/main/.github/workflows/build-test.yml) as example)

Reply:
I don't know if we will gain much in doing that. Those tasks run rather fast, at least headerCheck and formatting checks are fast. 

If we split, we will have to start different jobs, check out, sbt, etc. That may be too much overhead. 

Eventually, we can move the mimaCheck to the compilation task. Since we need to compile to check mima.